server/chat.ts

```diff
@@ -1647,6 +1652,18 @@ export const Chat = new class {
 			entry.hasRoomPermissions = /\bthis\.can\([^,)\n]*, [^,)\n]*,/.test(handlerCode);
 			entry.broadcastable = /\bthis\.(?:canBroadcast|runBroadcast)\(/.test(handlerCode);
 
+			// assign properties from the base command if the current command uses CommandContext.run.
+			const runsCommand = /this.run\((?:'|"|`)(.*?)(?:'|"|`)\)/.exec(handlerCode);
+			if (runsCommand) {
+				const [, baseCommand] = runsCommand;
+				const baseEntry = commandTable[baseCommand];
+				if (baseEntry) {
+					entry.requiresRoom = baseEntry.requiresRoom;
+					entry.hasRoomPermissions = baseEntry.hasRoomPermissions;
+					entry.broadcastable = baseEntry.broadcastable;
```

Comment:
```suggestion
					entry.requiresRoom = baseEntry.requiresRoom;
					entry.hasRoomPermissions = baseEntry.hasRoomPermissions;
					entry.broadcastable = baseEntry.broadcastable;
```

These should all be `||=` but sadly sucrase can't compile it yet.